x-pack/plugins/alerting/server/rules_client/lib/mapped_params_utils.ts

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { camelCase, snakeCase } from 'lodash';
+import { AlertTypeParams, MappedParams } from '../../types';
+import { SavedObjectAttribute } from '../../../../../../src/core/server';
+
+const FILTER_PARAMS_PREFIX = 'alert.attributes';
+
+export const MAPPED_PARAMS_PROPERTIES = ['risk_score', 'severity'];
+
+/**
+ * Returns the mapped_params object when given a params object.
+ * The function will match params present in MAPPED_PARAMS_PROPERTIES and
+ * return an empty object if nothing is matched.
+ */
+export const getMappedParams = (params: AlertTypeParams) => {
+  return Object.keys(params).reduce<MappedParams>((result, key) => {
+    const snakeCaseKey = snakeCase(key);
+
+    if (MAPPED_PARAMS_PROPERTIES.includes(snakeCaseKey)) {
+      result[snakeCaseKey] = params[key] as SavedObjectAttribute;
+    }
+
+    return result;
+  }, {});
+};
+
+/**
+ * Returns a string of the filter, but with params replaced with mapped_params.
+ * This function will check both camel and snake case to make sure we're consistent
+ * with the naming
+ *
+ * i.e.: 'alerts.attributes.params.riskScore' -> 'alerts.attributes.mapped_params.risk_score'
+ */
+export const getModifiedFilter = (filter: string | undefined) => {
```

Comment:
I will recommend to use the same technic to use KeuryNode to modify the filter as here -> https://github.com/elastic/kibana/blob/main/x-pack/plugins/alerting/server/rules_client/lib/validate_attributes.ts